### PR TITLE
UT for container (vector & map) operators (+,-,*)

### LINKED ISSUE #769: ABACUS lack unit test for container operators in container_operator.h (module_base)
## Background
ABACUS lack unit test for container operators in container_operator.h (module_base).

## Describe the solution you'd like
We should add unit test for  container operators in container_operator.h (module_base).

### PR DESCRIPTION
**Pull request by hongriTianqi**

### CODE REVIEW

**Comment by caic99** [comment]
The implementation of those "container operator" is low-efficient for cases with large data, and lacks checking on container size. Could you tell me where these methods are used in abacus? @hongriTianqi

File: source/module_base/test/container_operator_test.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+TEST(ContainerOperator,MapMinus)
+{
+	std::map<int,double> a;
+	std::map<int,double> b;
+	std::map<int,double> c;
+	for (int i=0;i<10;i++)
+	{
+		a.insert(std::pair<int,double> (i, i*4.0));
```
Comment (caic99) [inline source/module_base/test/container_operator_test.cpp:103]: What if `a` and `b` are of inequal length?

**Review by caic99 — CHANGES_REQUESTED** [review]

**Comment by hongriTianqi** [comment]
> 

The container operators are used in abacus-develop/source/src_ri/exx_abfs-dm.cpp @caic99

File: source/module_base/test/container_operator_test.cpp (hunk 1 above)
Comment (hongriTianqi) [reply source/module_base/test/container_operator_test.cpp:103]: They must be of the same length, which is not guaranteed in the current code. How shall we do this?

Comment (hongriTianqi) [reply source/module_base/test/container_operator_test.cpp:103]: assertions have been added

File: source/module_base/test/container_operator_test.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+{
+	std::vector<double> a(20,10.0);
+	double b = 2.0;
+	std::vector<double> c(20);
+	c = b*a;
+	for (int i=0;i<c.size();i++)
+		EXPECT_EQ(c[i],20.0);
+}
+
+TEST(ContainerOperator,LengthCheck)
```
Comment (caic99) [inline source/module_base/test/container_operator_test.cpp]: The name `MapLengthCheck` is better.
Is there any length check on vector operations?

**Review by caic99 — CHANGES_REQUESTED** [review]

**Review by caic99 — APPROVED** [review]

**Pull request merged**